watchtower/__init__.py

```diff
@@ -9,23 +11,46 @@
 from botocore.exceptions import ClientError
 
 
-def _idempotent_create(client, method, *args, **kwargs):
+if TYPE_CHECKING:
```

Comment:
Why is this guarded by a variable?

Reply:
Because `typing_extensions` is not required for the library, but when type checking it is expected (mypy depends on it itself so it will exist). `typing_extensions` is needed because protocol types weren't added until Python 3.8 https://docs.python.org/3/library/typing.html#typing.Protocol

Reply:
OK, thanks for the explanation. We will not be providing any type information for libraries other than this one, so please remove this whole block.